src/toxcore/dht/server/mod.rs

```diff
@@ -422,17 +429,43 @@ impl Server {
             )
         }
 
+        let mut futures = Vec::new();
+
+        if self.is_ipv6_enabled {
```

Comment:
Maybe add method to `DhtNode` that takes `is_ipv6_enabled` and returns `Vec<SocketAddr>`? Then we won't need to duplicate this logic here and in `send_to_node` method.